embassy-stm32/src/spi/mod.rs

```diff
@@ -178,6 +178,23 @@ impl<'d, T: Instance, Tx, Rx> Spi<'d, T, Tx, Rx> {
         )
     }
 
+    pub fn new_neopixel_driver(
```

Comment:
- can you make it consistent with nrf, as I pointed out in the issue? https://github.com/embassy-rs/embassy/issues/1343#issuecomment-1501211790
- If it's txonly it needs no rxdma.